### ISSUE TITLE
Conflict when used in combination with Vuetify

### ISSUE THREAD

**Issue by JenteVH** [jpex-js/vue-inject#16]
Hi,

When using vue-inject in combination with Vuetify there is a conflict resulting in an error.

Here is a repo which shows the problem
https://github.com/JenteVH/vuetify-bugchase

This is where the error is announced:
https://github.com/jpex-js/jpex/blob/master/src/resolver/factories.js#L28

Comments from other people:
"It looks like it hijacks directives and mixins for DI resolving as well, which leads to the problem"

**Comment by jackmellis** [comment]
Yeah I'll look into this. I think vue-inject makes too many assumptions about whether directives/mixins are injectable.

I don't think people use vue-inject for mixins very much. I'll probably add some configuration options to enable/disable the mixins/directives and disable them by default.

**Comment by robodude** [comment]
I've had issues with my mixins using dependencies. I had to write a custom merge strategy. It looks like this

```
var strategies = Vue.config.optionMergeStrategies
strategies.dependencies = function (toVal, fromVal) {
    if (!toVal) return fromVal
    if (!fromVal) return toVal
    return _.uniq([].concat(toVal, fromVal))
}
```

**Comment by jackmellis** [comment]
@JenteVH  I've fixed this issue in version 2.0.0

@robodude I've also added a merging strategy if one isn't already set
